_posts/2018-11-29-blog-build-chrome-ringcentral-widgets-extension-for-crms.markdown

```diff
@@ -0,0 +1,77 @@
+
+- For CRM contact list, we will add a hover-to-show tooltip to show click-to-call button.
+
+![ ](https://github.com/zxdong262/ringcentral-embeddable-extension-factory/raw/master/screenshots/fac-2.png)
+
+- For CRM contact info page, we will add a click-to-call button in proper position.
+
+![ ](https://github.com/zxdong262/ringcentral-embeddable-extension-factory/raw/master/screenshots/fac-3.png)
+
+- For CRM contact phone number text, we make it click-to-call link.
+
+![ ](https://github.com/zxdong262/ringcentral-embeddable-extension-factory/raw/master/screenshots/fac-1.png)
+
+- Sync CRM contacts to our widgets after user authorize.
+
+![ ](https://github.com/zxdong262/insightly-embeddable-ringcentral-phone/raw/master/screenshots/insightly-4.png)
+
+- Sync call log to CRM automatticly or mannually.
+
+![ ](https://github.com/zxdong262/hubspot-embeddable-ringcentral-phone/raw/master/screenshots/hs6.png)
+
+- Check CRM conatct activities from our widgets.
+
+![ ](https://github.com/zxdong262/hubspot-embeddable-ringcentral-phone/raw/master/screenshots/hs7.png)
+
+- Show CRM contact info panel when inbound/outbpund call with CRM contact.
+
+![ ](https://github.com/zxdong262/hubspot-embeddable-ringcentral-phone/raw/master/screenshots/hubspot1.png)
+
+## How do we do it
+
+Ok, two steps, build a chrome extension, and add all the above features, with [Embbnux Ji](https://github.com/embbnux)'s tuturial:
+ [Building Chrome Extension Integrations with RingCentral Embeddable](https://medium.com/ringcentral-developers/build-a-chrome-extension-with-ringcentral-embeddable-bb6faee808a3)
+and [third-party-service-in-widget document](https://github.com/ringcentral/ringcentral-embeddable/blob/master/docs/third-party-service-in-widget.md)
+
+Not hard, but still need more effort, so I created the [ringcentral-embeddable-extension-factory](https://github.com/zxdong262/ringcentral-embeddable-extension-factory) to minimize the work.
+
+We could simplify it like this.
+
+```bash
+# make sure you have npm@5.2+ installed
+npx ringcentral-embeddable-extension-factory my-app
+# or install it first
+# npm i -g ringcentral-embeddable-extension-factory && reef my-app
+# then carefully answer all questions, then the my-app folder will be create
+cd my-app
+npm i
+npm start
+# Then just follow my-app/README.md's instruction
+```
+
+![ ](https://github.com/zxdong262/ringcentral-embeddable-extension-factory/raw/master/screenshots/cli.png)
+
+For more detail, read [https://github.com/zxdong262/ringcentral-embeddable-extension-factory](https://github.com/zxdong262/ringcentral-embeddable-extension-factory)
+
+Still, this is just a skeleton, need developer efforts to make it a work, CLI tool can not know what css selector to use yet, do the crm support oauth or not, how to get the csrf token etc, not that smart. I do add some common module to do common task, developer could just set proper config to make it work.
```

Comment:
Try re-word it to sound more formal. Properly break a long sentence into short sentences that are too the point.